src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
If we could get a consensus, I'll do that. Now I have a tricky way to pass the validation in my 2.x project:
```php
public function isDateTime($check) {
    $value = array_values($check)[0];
    $value = implode(' ', explode('T', $value, 2));
    return Validation::datetime($value);
}
```
But it would be nice if Cake supports modern HTML features without this kind of trick.